### ISSUE TITLE
Possible in DClass belonging check bug

### ISSUE THREAD

**Issue by ZoltanCoccyx** [semigroups/Semigroups#931]
Hello !

I found this high strangeness behaviour:

```gap
LoadPackage("semigroups");

S := Monoid( [ Transformation( [ 3, 2, 3, 3, 5, 5 ] ), Transformation( [ 5, 4, 4, 5, 1, 4 ] ),  Transformation( [ 1, 1, 5, 3, 3 ] ), Transformation( [ 4, 5, 6, 4, 1, 4 ] ),  Transformation( [ 5, 1, 2, 1, 6, 5 ] ) ]);

e := Transformation( [ 1, 6, 6, 1, 5, 6 ] );
c := Transformation( [ 6, 1, 1, 6, 5, 1 ] );

Print(c in DClass(S, e), "\t", c in HClass(S, e));
```

On my installation, this prints "false true", which seems abnormal I think? Can you reproduce this ?

Thanks for your attention,

Best,

Balthazar

**Comment by james-d-mitchell** [comment]
Thanks @ZoltanCoccyx I'm investigating now!

**Comment by james-d-mitchell** [comment]
@ZoltanCoccyx definitely a bug, fix is in #932

**Comment by ZoltanCoccyx** [comment]
I do not detect any related problem anymore :)
